content/docs/for-buildpack-authors/how-to/write-buildpacks/specify-env.md

```diff
@@ -3,15 +3,160 @@ title="Specify the environment"
+
+* For layer path environment variables, user-provided values are prepended before any existing values and are delimited by the OS path list separator.
+* For all other environment variables, user-provided values override any existing values.
+* The environment variable prefix `CNB_` is reserved. It MUST NOT be used for environment variables that are not defined in this specification or approved extensions.
+
+### POSIX Path Variables
+
+When the `lifecycle` runs buildpacks, it first tears down anything defined on the `build-time` base image of the environment. It only allows a [specific set](https://github.com/buildpacks/lifecycle/blob/a43d5993a4f2cc23c44b6480ba2ab09fe81d57ed/env/build.go#L9-L19) of pre-configured environment variables through.
+
+Then, it applies `buildpack defined` environment variables including anything that might have been configured by a buildpack that ran earlier during the `build` phase.
+
+>Note that buildpacks cannot set environment variables for other buildpacks during the `detect` phase.
+
+#### Example
+
+Looking at the following directory tree and table
+
+```text
+
+layers/
+└── some-buildpack-id
+    ├── some-build-layer
+    │   ├── bin
+    │   │   └── some-binary
+    │   ├── env
+    │   │   └── SOME_VAR # contents foo
+    │   ├── env.build
+    │   │   └── SOME_VAR # contents bar
+    │   └── lib
+    │       ├── some-shared-library
+    │       └── some-static-library
+    └── some-build-layer.toml
+
+```
+
+| Env Variable                               | Layer Path   | Contents         | Build | Launch |
+|--------------------------------------------|--------------|------------------|-------|--------|
+| `PATH`                                     | `/bin`       | binaries         | [x]   | [x]    |
+| `LD_LIBRARY_PATH`                          | `/lib`       | shared libraries | [x]   | [x]    |
+| `LIBRARY_PATH`                             | `/lib`       | static libraries | [x]   |        |
+| `CPATH`                                    | `/include`   | header files     | [x]   |        |
+| `PKG_CONFIG_PATH`                          | `/pkgconfig` | pc files         | [x]   |        |
+
+* To add a directory to the $PATH variable for buildpacks that follow, we need to create a `<layers>/<layer>/bin` directory and add any executables we want to make available within that directory. The `lifecycle` will automatically add the new `bin` directory to the `PATH` for buildpacks that follow.
+* Binaries, such as `some-binary`, are added to `PATH` for buildpacks that follow
+* Shared libraries, such as `Some-shared-library` and `some-static-library`, are added to `LD_LIBRARY_PATH` and `LIBRARY_PATH`
+* If `env/SOME_VAR` and `env.build/SOME_VAR` have a conflict, a given procedure applies to figure out who "wins" when applying the environment [modification rules](#environment-variable-modification-rules).
```

Comment:
Maybe we don't need to talk about this part (too complicated, authors can consult the spec). Maybe we should just mention (at the very end) that it's possible to specify "build time only" and "launch time only" env vars (relevant when a layer is of type build AND launch).